va/validation-authority.go

```diff
@@ -447,24 +453,30 @@ func (va *ValidationAuthorityImpl) getCAASet(hostname string) (*CAASet, error) {
 			break
 		}
 
-		// Query CAA records for domain and its alias if it has a CNAME
-		for _, alias := range []bool{false, true} {
-			if alias {
-				target, _, err := va.DNSResolver.LookupCNAME(queryDomain)
-				if err != nil {
-					return nil, err
-				}
-				queryDomain = target
-			}
-			CAAs, _, err := va.DNSResolver.LookupCAA(queryDomain)
+		cnames := 0
+		for {
```

Comment:
Agreed the behavior is a little wonky (and I totally missed that we need to do DNAME as well, thanks for pointing that out) I think the reasoning is that while it is *invalid* DNS servers will still let you do it, so to avoid the path where a attacker is able to append a CNAME/DNAME but not delete/alter the CAA record the latter query is always performed first?